packages/search/addon/services/navi-search/navi-report-search-provider.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * This service is used to search for reports and dashboards stored in the persistence layer.
+ */
+
+import { inject as service } from '@ember/service';
+import NaviBaseSearchProviderService from '../navi-base-search-provider';
+
+export default class NaviReportSearchProviderService extends NaviBaseSearchProviderService {
+  /**
+   * @property {Ember.Service} store
+   */
+  @service store;
+
+  /**
+   * @property {Ember.Service} user
+   */
+  @service user;
+
+  /**
+   * @property {String} displayComponentName
+   */
+  displayComponentName = 'navi-search-result/report';
+
+  /**
+   * @method _parseReportString
+   * @private
+   * @param {String} query
+   * @param {String} type
+   * @returns {Object} query object
+   * @description Parses string query to search parameters
+   */
+  _parseQueryString(query, type) {
+    let author = this.user.getUser().id;
+    let parsedQuery = { searchParams: null, author: author };
+
+    if (typeof query == 'string' && query) {
+      if (type === 'report') {
+        parsedQuery.searchParams = {
+          title: query,
+          request: query
+        };
+      } else if (type === 'dashboard') {
+        parsedQuery.searchParams = {
+          title: query
+        };
+      }
+    }
+
+    return parsedQuery;
+  }
+
+  /**
+   * @method _constructSearchQuery
+   * @private
+   * @param {Object} searchParams
+   * @param {String} author
+   * @returns {Object} search query object
+   * @description Constructs the query filter parameters adhering to the RSQL standard
+   */
+  _constructSearchQuery(searchParams, author, type) {
+    let query = { filter: { [type]: '' } };
+
+    if (searchParams) {
+      for (let p in searchParams) {
```

Comment:
Instead of adding each character one by one could we map on the keys in searchParams, construct an array of filter strings for each param and call join on the array with a comma, then wrap that joined string in parentheses?

Reply:
```js
const paramsFilter = `${Object.keys(searchParams).map(p => `${p}==*${searchParams[p]}*`).join(',')}`; // comma separated list of param filters
const paramsFilterString = filters ? `(${filters})` : ''; //wrap in parentheses if param filter present
let authorFilterString = '';
if (author) {
  authorFilterString = paramsFilterString ? `;author==*${author}*` : `author==*${author}*`; //add semicolon if param filters present
}
const filterString = `${paramsFilterString}${authorFilterString}`;
query.filter[type] = filterString;
```